pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -79,272 +107,246 @@ func newPriorityAssigner(initialOFPriorityFunc InitialOFPriorityGetter, isSingle
-		p, _ := pa.ofPriorityMap[i+1]
-		pa.updatePriorityAssignment(i, p)
-		priorityReassignments[i+1] = i
-		klog.V(4).Infof("Original priority %v now needs to be re-assigned %v", i+1, i)
-	}
-	pa.updatePriorityAssignment(insertionPoint, p)
-	revertFunc := func() {
-		delete(pa.priorityMap, p)
-		for original := insertionPoint; original > lastVacant; original-- {
-			updated := original - 1
-			p := pa.ofPriorityMap[updated]
-			// lastVacant was allocated because of the reassignment and needs to be released when reverting.
-			if updated == lastVacant {
-				pa.Release(updated)
+		// all reassigned Priorities need to be assigned back to the original ofPriorities.
+		for p, update := range allPriorityUpdates {
+			if of, ok := pa.priorityMap[p]; ok {
+				pa.deletePriorityMapping(of, p)
 			}
-			pa.ofPriorityMap[original] = p
-			pa.priorityMap[p] = original
+			pa.ofPriorityMap[update.Original] = p
+			pa.priorityMap[p] = update.Original
 		}
 	}
-	return &insertionPoint, priorityReassignments, revertFunc, nil
-}
-
-// GetOFPriority retrieves the OFPriority for the input Priority to be installed,
-// and returns installed priorities that need to be re-assigned if necessary.
-func (pa *priorityAssigner) GetOFPriority(p types.Priority) (*uint16, map[uint16]uint16, func(), error) {
-	if ofPriority, exists := pa.priorityMap[p]; exists {
-		return &ofPriority, nil, nil, nil
-	} else if uint16(len(pa.sortedOFPriorities)) == PolicyTopPriority-PolicyBottomPriority+1 {
-		return nil, nil, nil, fmt.Errorf("priorities are fully registered on this table")
-	}
-	insertionPoint, occupied := pa.getInsertionPoint(p)
-	if insertionPoint == PolicyBottomPriority || insertionPoint > PolicyTopPriority || occupied {
-		return pa.reassignPriorities(insertionPoint, p)
+	for _, priorities := range consecutivePriorities {
+		if err := pa.insertConsecutivePriorities(priorities, allPriorityUpdates); err != nil {
+			// if failure occurred at any point, revert Priorities registered so far.
+			revertFunc()
+			return nil, nil, err
+		}
 	}
-	pa.updatePriorityAssignment(insertionPoint, p)
-	return &insertionPoint, nil, nil, nil
+	return priorityUpdatesToOFUpdates(allPriorityUpdates), revertFunc, nil
 }
 
-// RegisterPriorities registers a list of Priorities to be created with the priorityAssigner.
-// It is used to populate the priorityMap in case of batch rule adds. Note that this function assumes
-// that the pa.ofPriorityMap is empty at the time when it is called.
-func (pa *priorityAssigner) RegisterPriorities(priorities []types.Priority) error {
-	if len(pa.ofPriorityMap) != 0 {
-		return fmt.Errorf("cannot batch register priorities with a non empty priority map")
+// insertConsecutivePriorities inserts a list of consecutive Priorities into the ofPriority space.
+// It first identifies the lower and upper bound for insertion, by obtaining the ofPriorities of
+// registered Priority that is immediately lower and higher than the inserting Priorities. It then
```

Comment:
do you mean "by obtaining the ofPriorities of the registered Priorities surrounding (immediately lower and higher) the Priorities being inserted"?

Reply:
Thanks for the suggestion. Done.